Adds use cases and graphics to home page

ajessup opened the pull request:
This PR updates the homepage with a few tweaks to make it more approachable to new users. Specific changes include:

(1) Changing the project tagline to be more focused on the goals (rather than the history) of the project

(2) Introducing three broad use cases for SPIFFE and SPIRE, to help users understand if and how the project may be relevant to them

Preview build here: https://deploy-preview-169--spiffe.netlify.app/

evan2645 commented:
I haven't really read through your updates in depth, but they are looking great on the front page!

Maybe we should have some words about interoperability in the top 1/2 of the page? The current layout does a great job at conveying the problems we can solve for, but doesn't mention anything vision-wise e.g. "a standard authentication mechanism for ..."

This line also feels like it limits the audience to cloud native people:
```Industry best practice authentication for cloud-native services.```

anvega commented:
Thanks @ajessup!

I agree with @evan2645. SPIFFE and SPIRE aim to solve these problems in a common way that can be leveraged across many different types of systems -- not just containerized or cloud native. I also think "recommended" practice may be a less subjective term than "best" practice.

anvega commented on layouts/partials/home/use-cases.html:
```diff
@@ -0,0 +1,36 @@
+              <img src="/img/usecases/uc1-secure.svg">
+            </a>
+        </p>
+        <p>Secure microservices automatically with Envoy, X.509 PKI or JWT</p>
```
```suggestion
        <p>Automatically establish trusted bi-directional communication with Envoy, X.509 PKI or JWT</p>
```

ajessup replied:
I'm open to changes here but it's not clear to me what problem you're trying to solve with this proposal. FWIW this kind of language is generally successful when it speaks to a benefit (eg. "become more secure"), before describing a feature in detail.

I have concerns with the language - "Automatically establish trusted bi-directional communication" as being both jargon-heavy for a home page and opening up unnecessary questions for a new user. What does "establishing communication" mean? Is it a networking tool? Neither SPIFFE nor SPIRE facilitate "communication" per se. Communication between what? users? apps? What does "trusted" mean to a new user to SPIFFE? Trusted by who?

There's prior art for "secure" here as a shorthand establishing strong bi-directional authentication, eg. https://istio.io/  https://aspenmesh.io/

anvega replied:
I find the term secure to be loose and vague, and I'd rather have a higher degree of specificity on what is being secured, and how is being secured. 

In this case it is the traffic between two microservices as opposed to the microservices directly. 

Perhaps securing microservices communication, or establishing secure microservices connectivity. 

Also considering Evan'ss point, the applicability is not just limited to microservices but many different types of systems.

anvega commented on layouts/partials/home/use-cases.html:
```diff
@@ -0,0 +1,36 @@
+              <img src="/img/usecases/uc3-mesh.svg">
+            </a>
+        </p>
+        <p>Build, bridge and extend service mesh across organizations without sharing keys</p>
```
```suggestion
        <p>Build, bridge and extend service mesh across technology boundaries without sharing keys</p>
```

ajessup replied:
The goal of this copy was to indirectly highlight the utility of federation as a technology that can allow trusted interop between different groups that may wish to securely authenticate (kinda to Evan's point above) without trusting a common key. Your proposal changes the meaning of this to focus on deployment in heterogenous environments instead. I'd like to see if there's a better way of doing that elsewhere in the homepage.

anvega replied:
Thanks for the explanation. I now see where you are going with it. 

The point is valid in the case of both organizations in the sense of administrative boundaries as you describe it as well as technological boundaries, or boundaries alone, whatever those may be.

anvega commented on config.toml:
```diff
@@ -18,7 +18,7 @@ githuburl = "https://github.com/spiffe/spiffe.io/tree/master"
 [params.home]
 acronym     = "**S**ecure **P**roduction **I**dentity **F**ramework **f**or **E**veryone"
-description = "Inspired by the production infrastructure of Google and others, SPIFFE is a set of open-source standards for securely identifying software systems in dynamic and heterogeneous environments. "
+description = "Industry best practice authentication for cloud-native services. "
```
How about:

```suggestion
description = "Production identity control plane for distributed systems. "
```

ajessup replied:
I'm open to changes here, and I like "identity control plane" as a description generally but I'm not comfortable that a new user to SPIRE would intuitively understand what an "identity control plane" is (and thus why they would want one). This might be something to user test or UX test before changing.

anvega replied:
There is not much that a new user will understand intuitively but at least they are in the right place. Here lies a unique opportunity to create a clear story, define a category blueprint, and shape the thinking of new users. 

The Benefits and use cases described below tie the concept together by giving them what an identity control plane is and why the would want one. 

New users will have arrived at this page knowing at least something. In the worst case, every single new user has some background knowledge of software or systems. They will know what a control plane is.

ajessup replied:
Update: From the discussion below it sounds like this resonates. Let's try it :)

anvega commented:
Perhaps a  "has-text-centered use-cases" section in white background color might look better than grey so that the logo and use cases show as one section.

anvega commented on config.toml:
```diff
@@ -18,7 +18,7 @@ githuburl = "https://github.com/spiffe/spiffe.io/tree/master"
+description = "Industry best practice authentication for cloud-native services. "
 
 [params.home.whatis]
 spiffe = "**SPIFFE**, the Secure Production Identity Framework For Everyone, provides a secure identity, in the form of a specially crafted X.509 certificate, to every workload in a modern production environment. SPIFFE removes the need for application-level authentication and complex network-level ACL configuration. "
```
```suggestion
spiffe = "**SPIFFE**(Secure Production Identity Framework For Everyone) is a building block specification for doing identity distribution and attestation for service to service communication.  Think of it as, roughly, a cert distribution protocol. 

SPIFFE is very well aligned with the ideas behind _zero-trust networks._  The goal is to provide a way for a program to run and, from its environment, get all the information necessary to communicate securely over the network."
```

anvega replied:
```suggestion
spiffe = "**SPIFFE**(Secure Production Identity Framework For Everyone) is a building block specification for doing identity distribution and attestation for service to service communication.  Think of it as, roughly, a cert distribution protocol. 
SPIFFE is very well aligned with the ideas behind _zero-trust networks._  The goal is to provide a way for a program to run and, from its environment, get all the information necessary to communicate securely over the network."
```

sanderson042 commented:
I'll let y'all hash out the wording, but I'd prefer using the [serial comma](https://en.wikipedia.org/wiki/Serial_comma) because it's the most common approach in American English. For example:

_Automatically establish trusted bi-directional communication with Envoy, X.509 PKI, or JWT_

umairmkhan commented:
I think we tried "Identity plane for cloud-native services" or "Cloud-native identity plane for software services" at shows it worked and was easier to explain plus I think talking about identity evaluates it beyond just authentication (which can be covered by use cases.

umairmkhan commented:
For use cases I think we can expand on these a bit and perhaps link them to sections in docs? I can work with Steve this week to make some suggestions.

ajessup commented:
Thanks for the feedback folks, appreciated!  The motivation behind this PR was try and to simplify decision making for a potentially new user of SPIFFE/SPIRE. To do this successfully the above-the-fold copy of this page should be able to answer, reasonably succinctly: 

* What problems that I know I have can this project solve for me?
* Why is this project specifically the best solution to those problems for me? What makes it special?

> Maybe we should have some words about interoperability in the top 1/2 of the page? The current layout does a great job at conveying the problems we can solve for, but doesn't mention anything vision-wise e.g. "a standard authentication mechanism for ..."

The avoidance of a vision statement was a deliberate choice, on the assumption that new users will be more concerned with what they _can_ do with the project right now than what the project eventually aspires to. The latter can be disheartening if a new user mistakes a statement of vision for a claim of capability, and then finds the project comes up short.
 
FWIW, I think a statement of principles/design goals/aspirations - particularly around interop - for the project somewhere on our website is a great idea - I'm just not sure it's something we want to highlight on our landing page unless it translates into a concrete and unique benefit that actually accrues to a user of the product. 

As it stands, I'm not sure "interop" meets that bar today.

If you mean interop between different workload types (containers vs. functions, for example?). If so that's certainly possible today, but more or less any authentication standard (eg. oAuth, Kerberos) can give you that, so not sure it makes sense to highlight it as a benefit.

If you mean a standard way for a workload to retrieve its identity (ie. the Workload API)? If so we should make sure this is an ideal we can actually hold to - today most SPIFFE implementations _don't_ support the Workload API, and the one that does (SPIRE) may well not be able to when it operates in an agent-less environment like Lambda or Fargate.

If you mean interop between different implementation/installations of SPIFFE issuing authorities (eg. Istio to Consul to SPIRE)? If so I agree this is a great benefit to highlight, once there's enough alignment between implementations to make it a true statement.

> 
> This line also feels like it limits the audience to cloud native people:
> `Industry best practice authentication for cloud-native services.`

I think the language we aligned on above now satisfies this while avoiding "cloud native".

ajessup commented:
@umairmkhan I love the idea of linking to sections in the docs that elaborate on each of the use cases. I would suggest we do that in a separate PR once we've aligned on the language here.

evan2645 commented:
> ... the assumption that new users will be more concerned with what they can do with the project right now than what the project eventually aspires to. The latter can be disheartening if a new user mistakes a statement of vision for a claim of capability, and then finds the project comes up short.

Fair point!

> FWIW, I think a statement of principles/design goals/aspirations - particularly around interop - for the project somewhere on our website is a great idea - I'm just not sure it's something we want to highlight on our landing page unless it translates into a concrete and unique benefit that actually accrues to a user of the product.

What motivates a user to choose SPIFFE over any of the other bazillion choices they have for identity and authentication? Certainly, many of them are easier to adopt and use than SPIFFE (e.g. k8s service account token, which comes "for free"). I make the interop comment because IMO that is the really big differentiator. You have a good point that maybe we do not fully meet the "interop" vision today, but it still feels important to signal this aspect. Perhaps "an open standard" or similar language can address it? Why go through the trouble to use SPIFFE when there are so many other options? We don't want to be just another thing.

ajessup commented:
> What motivates a user to choose SPIFFE over any of the other bazillion choices they have for identity and authentication? Certainly, many of them are easier to adopt and use than SPIFFE (e.g. k8s service account token, which comes "for free"). I make the interop comment because IMO that is the really big differentiator. You have a good point that maybe we do not fully meet the "interop" vision today, but it still feels important to signal this aspect. Perhaps "an open standard" or similar language can address it? Why go through the trouble to use SPIFFE when there are so many other options? We don't want to be just another thing.

I take your point. Perhaps we can incorporate this idea into the tagline, eg.

"A universal production identity control plane for distributed systems."

or

"A cross-platform production identity control plane for distributed systems."

evan2645 commented:
> "A universal production identity control plane for distributed systems."

I think that does it!

I prefer it over the second suggestion as many existing solutions are `cross-platform` by nature due operation at the application layer

ajessup commented:
Thanks folks! @anvega @sanderson042 any further concerns? If not can I get this approved?

anvega commented:
I worry that defining the first use case "secure microservices automatically" can be seen as an overstatement and it would be better qualified as securing the communications between microservices. 

For consistency, the use cases section might look better in white background as opposed to grey to be closer in length to the other layout section in white further down the page.

sanderson042 commented:
Some nits:

To me, the "Secure Production Identity Framework for Everyone" competes for attention with the new tagline. If it was a lighter font or more gray that might fix it. Alternatively, the new tagline could be slightly bolder or more dark.

I don't feel strongly about it, but IMO we should omit the period at the end of the tagline:
_A universal identity control plane for distributed systems_

Also, let's add those serial commas to the text below the new graphics as I mentioned earlier. America! 🇺🇸

ajessup commented:
From @anvega:

> I worry that defining the first use case "secure microservices automatically" can be seen as an overstatement and it would be better qualified as securing the communications between microservices.

Good point. Fixed.

> 
> For consistency, the use cases section might look better in white background as opposed to grey to be closer in length to the other layout section in white further down the page.

Agreed. Fixed.

From @sanderson042:

> To me, the "Secure Production Identity Framework for Everyone" competes for attention with the new tagline. If it was a lighter font or more gray that might fix it. 

Fixed.

> I don't feel strongly about it, but IMO we should omit the period at the end of the tagline:

Fixed

> Also, let's add those serial commas to the text below the new graphics as I mentioned earlier. 

Fixed, I think ;) Sorry for missing this earlier.

anvega reviewed (approved):
👌
